refactor: LEAP-1772: Remove FF_OPTIC_2 feature flag and associated logic

Gondragos opened the pull request:
Eliminated the `fflag_feat_optic_2_ensure_draft_saved_short` feature flag and related conditional checks from the codebase as it is no longer needed.

| **Key** | `fflag_feat_optic_2_ensure_draft_saved_short` |
|---|---|
| **Variable Name** | `FF_OPTIC_2` |
| **Ends up as** | 🟩 `True` |
| **Description** | Save draft on leave annotation editor |

Gondragos committed:
refactor: Remove FF_OPTIC_2 feature flag and associated logic

Eliminated the `fflag_feat_optic_2_ensure_draft_saved_short` feature flag and related conditional checks from the codebase as it is no longer needed.

Gondragos committed:
fix: remove unused `useFixedLocation` and `deleteAction`

The `useFixedLocation` hook and `deleteAction` function calls were removed due to being unused.

Gondragos committed:
Merge branch 'develop' into fb-LEAP-1772/stale-ff

# Conflicts:
#	web/apps/labelstudio/src/pages/DataManager/DataManager.jsx

Gondragos committed:
fix: remove stale feature flag from list

Removed the flag 'fflag_feat_optic_2_ensure_draft_saved_short' from the stale feature flags list as it is no longer in use.

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13918755434)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-LEAP-1772/stale-ff'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13918755434

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13918755434

Gondragos commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13927161417)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13927161417

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13927307413)
> Unresolvable merge conflict detected in files:
label_studio/core/feature_flags/stale_feature_flags.py
Please resolve it using the git command line

Gondragos committed:
Merge branch 'develop' into fb-LEAP-1772/stale-ff

# Conflicts:
#	label_studio/core/feature_flags/stale_feature_flags.py
#	web/libs/datamanager/src/utils/feature-flags.js

Gondragos commented on label_studio/core/feature_flags/stale_feature_flags.py:
```diff
@@ -2,6 +2,7 @@
     'fflag_feat_all_dia_13_structured_data_support_short': False,
     'fflag_feat_front_optic_66_lazy_chart_evaluation_19092023_short': False,
     'fflag_feat_optic_2_ensure_draft_saved_short': True,
+    'fflag_feat_front_lsdv_4620_outliner_optimization_310723_short': True,
```
```suggestion
```

Commit:
Revert wrong merge result

Gondragos committed:
fix: remove stale feature flag once again

Gondragos committed:
Merge remote-tracking branch 'origin/fb-LEAP-1772/stale-ff' into fb-LEAP-1772/stale-ff

# Conflicts:
#	label_studio/core/feature_flags/stale_feature_flags.py

Gondragos committed:
fix: correct indentation in stale feature flags file
